engine/src/flutter/shell/platform/windows/flutter_windows_view.cc

```diff
@@ -214,19 +212,30 @@ bool FlutterWindowsView::OnWindowSizeChanged(size_t width, size_t height) {
     return true;
   }
 
-  resize_status_ = ResizeState::kResizeStarted;
-  resize_target_width_ = width;
-  resize_target_height_ = height;
+  {
+    std::unique_lock<std::mutex> lock(resize_mutex_);
+    resize_status_ = ResizeState::kResizeStarted;
+    resize_target_width_ = width;
+    resize_target_height_ = height;
+  }
 
   SendWindowMetrics(width, height, binding_handler_->GetDpiScale());
 
-  // Block the platform thread until a frame is presented with the target
-  // size. See |OnFrameGenerated|, |OnEmptyFrameGenerated|, and
-  // |OnFramePresented|.
-  return resize_cv_.wait_for(lock, kWindowResizeTimeout,
-                             [&resize_status = resize_status_] {
-                               return resize_status == ResizeState::kDone;
-                             });
+  std::chrono::time_point<std::chrono::steady_clock> start_time =
+      std::chrono::steady_clock::now();
+
+  while (true) {
+    if (std::chrono::steady_clock::now() > start_time + kWindowResizeTimeout) {
```

Comment:
Should we update this comment?

https://github.com/flutter/flutter/blob/5e37c966c04077fff16a48c045c71471be16739c/engine/src/flutter/shell/platform/windows/flutter_windows_view.cc#L21-L23

Maybe something like:

```diff
-// The maximum duration to block the platform thread for while waiting 
+// The maximum duration to block the Windows event loop while waiting 
 // for a window resize operation to complete.
```